arbor/backends/gpu/threshold_watcher.hpp

```diff
@@ -141,7 +141,7 @@ class threshold_watcher {
     array v_prev_;              // Values at previous sample time.
 
     // Hybrid host/gpu data structure for accumulating threshold crossings.
-    stack_type stack_;
+    mutable stack_type stack_;
```

Comment:
Why is this mutable?

Reply:
Because of the `const` functions that involve copying back and forth from GPU memory into a container that's a member of `stack_type`.